### PR TITLE
AB#52204 Added MtoaRequestService

### PR DESCRIPTION
**Pull request by tropical-turtle**

### CODE REVIEW

File: CDTS Blazor/Models/PageModels/ReviewDocumentPageModel.cs [hunk 1]
```diff
@@ -109,17 +109,13 @@ protected override void OnInitialized()
...
-
             if (this.State.UploadDocumentPage == null)
             {
                 this.State.UploadDocumentPage = new UploadDocumentPageModel();
             }
 
             this.Model = this.State.UploadDocumentPage;
-
-            this.Model.ConfirmationNumber = this.confirmationGuid.ToString();
+            this.confirmationGuid = this.Model.MtoaServiceRequestId;
```
Comment (AlexandreSanscartier) [inline CDTS Blazor/Models/PageModels/ReviewDocumentPageModel.cs:118]: Do we want the confirmationGuid to be the service request Id?

File: CDTS Blazor/Pages/UploadDocument.razor [hunk 2]
```diff
@@ -288,6 +289,9 @@
         certificateTypes = CertificateType.GetAllCertificateTypes();
 
         currentFileName = PageLocalizer["FileNamePlaceholder"];
+
+        model.MtoaServiceRequestId = requestService.GetServiceRequestIdFromMTOA();
+        model.ConfirmationNumber = model.MtoaServiceRequestId.ToString();
```
Comment (AlexandreSanscartier) [inline CDTS Blazor/Pages/UploadDocument.razor:294]: Do we want the confirmationGuid to be the service request Id?

File: CDTS Blazor/Utilities/RestClient.cs [hunk 3]
```diff
@@ -103,6 +103,21 @@ public async Task<TReturnMessage> PostAsync<TReturnMessage>(ServiceLocatorDomain
...
+        /// <returns>A <see cref="Task{HttpResponseMessage}"/> task of HttpResponseMessage.</returns>
+        public async Task<HttpResponseMessage> PostAsync(ServiceLocatorDomain serviceName, string path)
+        {
+            var uri = new Uri($"{this.serviceLocator.GetServiceUri(serviceName)}/{path}");
+
+            var response = await Client.PostAsync(uri, null).ConfigureAwait(false);
+
+            return response;
+        }
+
```
Comment (AlexandreSanscartier) [inline CDTS Blazor/Utilities/RestClient.cs:120]: Shouldn't need to add a new method to the RestClient
```suggestion
```

File: CDTS Blazor/Utilities/IRestClient.cs [hunk 4]
```diff
@@ -28,6 +29,14 @@ Task<TReturnMessage> GetAsync<TReturnMessage>(ServiceLocatorDomain serviceName,
...
             where TReturnMessage : class, new();
 
+        /// <summary>
+        /// Makes POST request with full Uri.
+        /// </summary>
+        /// <param name="serviceName">Name of the service as specified by <see cref="ServiceLocatorDomain"/>.</param>
+        /// <param name="path">Path to the API being called on the service.</param>
+        /// <returns>A <see cref="Task{HttpResponseMessage}"/> task of HttpResponseMessage.</returns>
+        public Task<HttpResponseMessage> PostAsync(ServiceLocatorDomain serviceName, string path);
+
```
Comment (AlexandreSanscartier) [inline CDTS Blazor/Utilities/IRestClient.cs:39]: Shouldn't need to add this method we can use the PostAsync(ServiceLocatorDomain serviceName, string path, object dataObject = null);
```suggestion
```

File: CDTS Blazor/appsettings.json [hunk 5]
```diff
@@ -42,7 +42,11 @@
     "SeafarerCertificationServiceName": "CSF",
     "ReplyEmail": "donotreply-nepasrepondre@tc.gc.ca",
     "ServiceRequestId": "13844",
-    "UserId": "4536"
+    "UserId": "4888",
+    "ServiceId": 57,
+    "ServiceNameInEnglish": "Seafarer certification",
+    "ServiceNameInFrench": "Certification des gens de mer",
+    "ProgressStatus": "InProgress"
```
Comment (AlexandreSanscartier) [inline CDTS Blazor/appsettings.json:49]: Is there a reason we're adding a progress status here? doesn't seem like the right place to put it.

**Review by AlexandreSanscartier — CHANGES_REQUESTED** [review]

File: CDTS Blazor/Models/PageModels/ReviewDocumentPageModel.cs (hunk 1 above)
Comment (AlexandreSanscartier) [reply CDTS Blazor/Models/PageModels/ReviewDocumentPageModel.cs:118]: Nvm we do lol

**Comment by tc-tibo** [comment]
The PR links to issue #91

**Pull request merged**